pkg/volume/volume.go

```diff
@@ -134,6 +134,15 @@ type MounterArgs struct {
 	DesiredSize         *resource.Quantity
 	SELinuxLabel        string
 	Recorder            record.EventRecorder
+
+	// Optional interface that will be used to change the ownership of the volume, if specified.
+	// mainly used by unit tests
+	VolumeOwnershipApplicator VolumeOwnershipChanger
```

Comment:
The MounterArgs struct is used to pass arguments to the Mounter interface for all volume types. And the current fields are populated by the `GenerateMountVolumeFunc` function in the operation_generator.go.It is possible to populate the new field in that function. And other volume types don't respect the new field like emptyDir, secret and etc. If the new field should only initialized in the `SetUp` function and not intent to be used in the `GenerateMountVolumeFunc` function, we should document that in the MounterArgs struct. Otherwise, we should add the new field to the `GenerateMountVolumeFunc` function and make sure all volume types respect it.

https://github.com/gnufied/kubernetes/blob/pr/gnufied/131311/pkg/volume/util/operationexecutor/operation_generator.go#L582

Reply:
>If the new field should only initialized in the SetUp function and not intent to be used in the GenerateMountVolumeFunc function, we should document that in the MounterArgs struct.

We already documented here that this is an optional field and only used in unit tests.  Is there any other way to document this?  

>Otherwise, we should populate the new field to the GenerateMountVolumeFunc function and make sure all volume types respect it.

I do not see any value in this doing at all. It is pretty normal to have an optional field that only certain providers will use or act on. What you are proposing will lead to changing all plugins for a field that they won't even use. Seems unnecessary.